### ISSUE TITLE
Uncaught Error: ELOOP: too many symbolic links

### ISSUE THREAD

**Issue by BroukPytlik** [v3ss0n/goto#74]
If I have a symlinks loop somewhere in the path scanned by goto, this plugin attempts to do an infinite recursion and is killed with ELOOP.

**Atom Version**: 1.2.4
**System**: "Fedora release
**Thrown From**: [goto](https://github.com/v3ss0n/goto) package, v1.7.1
### Stack Trace

Uncaught Error: ELOOP: too many symbolic links encountered, stat '/home/jtulak/some-project/include/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk'

```
At ATOM_SHELL_ASAR.js:0

Error: ELOOP: too many symbolic links encountered, stat '/home/jtulak/some-project/include/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk/disk'
    at Error (native)
    at fs.statSync (fs.js:854:18)
    at Object.fs.statSync (ATOM_SHELL_ASAR.js:225:16)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:168:18)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.processDirectory (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:178:8)
    at SymbolIndex.module.exports.SymbolIndex.rebuild (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:117:8)
    at SymbolIndex.module.exports.SymbolIndex.update (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:109:8)
    at SymbolIndex.module.exports.SymbolIndex.getAllSymbols (/home/jtulak/.atom/packages/goto/lib/symbol-index.coffee:100:6)
    at Object.module.exports.gotoProjectSymbol (/home/jtulak/.atom/packages/goto/lib/index.coffee:41:22)
    at atom-workspace.atom.commands.add.goto:project-symbol (/home/jtulak/.atom/packages/goto/lib/index.coffee:20:34)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:99:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```
### Commands

```
  2x -5:33.6.0 vim-mode-plus:delete (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.visual-mode.characterwise)
     -5:32.8.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.operator-pending-mode)
     -5:32.5.0 vim-mode-plus:undo (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -5:31.9.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
  2x -5:31.5.0 vim-mode-plus:delete (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -5:30.4.0 vim-mode-plus:undo (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -5:27.5.0 vim-mode-plus:put-after (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -5:27 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -5:24.6.0 core:save (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
  2x -5:22.3.0 blur (atom-text-editor.editor.vim-mode-plus.is-focused.reversed.normal-mode)
     -4:39.9.0 vim-mode-plus:activate-insert-mode (atom-text-editor.editor.vim-mode-plus.reversed.normal-mode.is-focused)
 15x -4:38.4.0 blur (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.insert-mode)
  3x -4:04.1.0 core:backspace (atom-text-editor.editor.vim-mode-plus.reversed.insert-mode.is-focused)
     -3:59.9.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.reversed.insert-mode.is-focused)
     -3:52.2.0 goto:project-symbol (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
     -0:56.8.0 blur (atom-text-editor.editor.vim-mode-plus.reversed.is-focused.normal-mode)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "ex-mode"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
Sublime-Style-Column-Selection, v1.3.0
goto, v1.7.1
minimap, v4.16.2
preview-inline, v0.5.1
vim-mode-plus, v0.7.1

# Dev
No dev packages
```

**Comment by BroukPytlik** [comment]
A simple way how to create a loop: ```ln -s . infinite_symlink```  (note the dot).

**Comment by acusti** [comment]
Should be fixed in 1.8.0
